pkg/logs/input/docker/client.go

```diff
@@ -1,82 +0,0 @@
-// Unless explicitly stated otherwise all files in this repository are licensed
-// under the Apache License Version 2.0.
-// This product includes software developed at Datadog (https://www.datadoghq.com/).
-// Copyright 2016-2020 Datadog, Inc.
-
-// +build docker
-
-package docker
-
-import (
-	"context"
-	"fmt"
-
-	"github.com/docker/docker/api/types"
-	"github.com/docker/docker/api/types/filters"
-	"github.com/docker/docker/api/types/versions"
-	"github.com/docker/docker/client"
-)
-
-// Supported versions of the Docker API
-const (
-	minVersion = "1.18"
-	maxVersion = "1.25"
```

Comment:
Are we sure that nothing will break without those two checks ? e.g. what happens if a user has a version prior to `1.18` with the new change ?

Reply:
Most of the `DockerUtil` services that are now used in the log agent code were already existing and were already used by some other parts of the agent.
The only function that I added to `DockerUtil` to support the use cases of the log agent is `ContainerLogs`:
https://github.com/DataDog/datadog-agent/blob/386a22ab3b0efdfc290d6667160d2b5625960736/pkg/util/docker/containers.go#L162-L164
And the API used to retrieve the logs is the same (same URL and same parameters) between the [oldest documented API version (1.18)](https://docs.docker.com/engine/api/v1.18/#get-container-logs) and the [newest documented API version (1.40)](https://docs.docker.com/engine/api/v1.40/#operation/ContainerLogs).

[The docker website is not documenting API versions prior to `1.18` anymore.](https://docs.docker.com/engine/api/version-history/)

The docker client that we vendor doesn’t document versions prior to `1.18` either:
```
~/devel/DataDog/datadog-agent $ ls vendor/github.com/docker/docker/docs/api
v1.18.md  v1.19.md  v1.20.md  v1.21.md  v1.22.md  v1.23.md  v1.24.md  version-history.md
```

Here is the PR that introduced this check: #1164.

Do you remember if there was an issue with docker API version prior to `1.18` or if we added this check just to avoid untested setup because we didn’t validate the agent with old docker daemon ?

Anyhow, before this change, if the agent tried to talk to a docker daemon which doesn’t support API 1.18 or above, we would fail the instantiation of the docker client. But it wouldn’t prevent the log agent from receiving new services and sources notification because the AD was already using `GetDockerUtil`.
With this change, the instantiation of the client would succeed. If the API version doesn’t support log retrieval, that’s the `ContainerLogs()` function that will return an error.